experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -202,6 +203,64 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
+	public reduce<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U,
+		initialValue?: U | undefined,
+	): UnboxNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => unboxedUnion(this.context, this.schema, cursor),
+			initialValue,
+		);
+	}
+
+	public reduceBoxed<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: TypedNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U,
+		initialValue?: U | undefined,
+	): TypedNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => unboxedUnion(this.context, this.schema, cursor),
+			initialValue,
+		);
+	}
+
+	private reduceImpl<Accumulation, T>(
```

Comment:
Do we need two types here? All usages I've seen the types are the same, and all having two of them gains you is preventing the callback from returning T if its not included in the Accumulation type. I don't think an extra generic type to prevent a ussage pattern is useful in a private helper.

Reply:
I made multiple mistakes here, so it didn't make a lot of sense as it was, sorry. The `T` is to accommodate both kinds of nodes (boxed and unboxed); I had made a copy-paste error so it was unboxed both times. I'm still unhappy with the cast, though, but I can't figure out how to do better (without maybe creating multiple overloads for this private helper, which I agree might be overkill).